Unity-Technologies/com.unity.multiplayer.samples.coop
# fix: disconnect issue with wifi turned off

### Description (*)
Disconnecting wifi caused an issue where clients wouldn't be able to reconnect.
This is due to a bug in NGO's Shutdown, which happens when it's called when NGO isn't Listening.

This PR adds a check around each Shutdown to make sure we're listing before calling it. --> workaround
This PR also cleans up our disconnect handling, to make sure we're using the same connection flow each time. --> cherrypick
This also fixes an issue where we didn't sent the appropriate allocation Id to lobby. --> cherrypick

### Related Pull Requests

### Issue Number(s) (*)

Fixes issue(s): [MTT-2833](https://jira.unity3d.com/browse/MTT-2833)
### Manual testing scenarios

client on local machine, host on different machine. Connect client and host together using relay. Disconnect client's wifi. Wait 10-15 seconds (screen should go black with some curl error messages saying unable to reach services). Reconnect wifi, client should be brought back to main menu. Host should show only 1 connected player. Try to connect client to host again. There will be a silent fail and client won't be able to connect.

In boss room, we try to reconnect you once if connection drops. In our case, since lobby requests fail, we never reach the "StartClient" part of our reconnection process. We then go to the "reconnection failed" part, which calls Shutdown on our network manager.
